iphone/TitaniumKit/TitaniumKit/Sources/API/WebFont.m

```diff
@@ -34,25 +34,10 @@ - (BOOL)isSizeNotSet
   return size < 4;
 }
 
-- (BOOL)isValidTextStyle:(NSString *)theStyle
-{
-  return (
-      [theStyle isEqualToString:UIFontTextStyleBody] ||
-      [theStyle isEqualToString:UIFontTextStyleCaption1] ||
-      [theStyle isEqualToString:UIFontTextStyleCaption2] ||
-      [theStyle isEqualToString:UIFontTextStyleHeadline] ||
-      [theStyle isEqualToString:UIFontTextStyleSubheadline] ||
-      [theStyle isEqualToString:UIFontTextStyleFootnote] ||
-      [theStyle isEqualToString:UIFontTextStyleCallout] ||
-      [theStyle isEqualToString:UIFontTextStyleTitle3] ||
-      [theStyle isEqualToString:UIFontTextStyleTitle2] ||
-      [theStyle isEqualToString:UIFontTextStyleTitle1]);
-}
-
 - (UIFont *)font
 {
   if (font == nil) {
-    if (textStyle != nil && [self isValidTextStyle:textStyle]) {
```

Comment:
Any particular reason you removed this check? What will happen if the text style is not one of the valid constants?

Reply:
I wanted to remove the restriction to allow future font-styles to be applied even if the constant is not exposed so far, e.g. a raw-value of `large-title` which can map to `UIFontTextStyleLargeTitle`. It does allow invalid values as well, but that's caught by the `UIFont` convenience constructor anyway, so double-checking this every time does not make much sense. In the end, it's the developers choice what do pass and if it's invalid, he/she/it will be informed.